server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -896,6 +896,25 @@ public Map<String, ComposableIndexTemplate> templatesV2() {
             .orElse(Collections.emptyMap());
     }
 
+    public boolean isTsdbTemplate(ComposableIndexTemplate indexTemplate) {
+        var template = indexTemplate.template();
+        if (indexTemplate.getDataStreamTemplate() == null || template == null) {
+            return false;
+        }
+
+        var settings = MetadataIndexTemplateService.resolveSettings(indexTemplate, componentTemplates());
+        if (IndexMetadata.INDEX_ROUTING_PATH.exists(settings)) {
```

Comment:
If we're going to auto-generate the routing path maybe we should use `index_mode` here?

Reply:
This method is going to change once `index.routing_path` is going to be generated from the template's mapping. Maybe we can also use `index_mode` index setting here, but that isn't something we said a user would need to configure in order to create data streams in tsdb mode.

Reply:
I *think* configuring the mode is still a good thing. Right now we support configuring fields as `time_series_dimension`s and without modifying the index itself. It'd be pretty surprising if upgrading Elasticsearch caused new indices with those mappings to be time_series. I'm not sure that's a terrible idea, but I think it's worth double checking. Personally I'd only try and generate the `routing_path` if the index is configured in `time_series` mode. But I might be wrong here.

Reply:
Either way, that concern doesn't block this PR - it'd block the PR that generates the `routing_path`.